mlir/lib/Conversion/RockToGPU/RockToGPU.cpp

```diff
@@ -181,6 +184,24 @@ void LowerRockOpsToGPUPass::runOnOperation() {
     b.setInsertionPointToEnd(&gpuFuncEntry);
     b.create<cf::BranchOp>(loc, clonedFuncEntry);
 
+    // Clone in global constants
+    WalkResult result =
+        funcBody.walk([&b, &gpuMod, &gpuModuleSymbolTable](
+                          memref::GetGlobalOp op) -> WalkResult {
+          SymbolRefAttr globalSym = op.getNameAttr();
+          auto toClone = dyn_cast_or_null<memref::GlobalOp>(
+              SymbolTable::lookupNearestSymbolFrom(op, globalSym));
+          if (!toClone)
+            return WalkResult::interrupt();
+          OpBuilder::InsertionGuard guard(b);
+          Operation *cloned = toClone.clone();
+          // There probably shouldn't be any renames, but let's be careful.
+          StringAttr newNameAttr = gpuModuleSymbolTable.insert(cloned);
+          op.setNameAttr(FlatSymbolRefAttr::get(newNameAttr));
```

Comment:
But now you will have to update references if the name changed. Probably best to assert that it didn't change.

Reply:
Ok, I've now tried to do something about that (and the possibility that we might clone the same constant twice)